vnext/ReactUWP/Modules/AppearanceModule.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "AppearanceModule.h"
+
+#include <winrt/Windows.UI.ViewManagement.h>
+
+using Application = winrt::Windows::UI::Xaml::Application;
+using ApplicationTheme = winrt::Windows::UI::Xaml::ApplicationTheme;
+using UISettings = winrt::Windows::UI::ViewManagement::UISettings;
+
+using Method = facebook::xplat::module::CxxModule::Method;
+
+namespace react::uwp {
+
+AppearanceChangeListener::AppearanceChangeListener(std::weak_ptr<IReactInstance> &&reactInstance) noexcept
+    : Mso::ActiveObject<>(Mso::DispatchQueue::MainUIQueue()), m_weakReactInstance(std::move(reactInstance)) {}
+
+void AppearanceChangeListener::Initialize() noexcept {
+  m_currentTheme = Application::Current().RequestedTheme();
+
+  // UISettings will notify us on a background thread regardless of where we construct it or register for events. Let it
+  // be constructed before init, but redirect callbacks to the UI thread where we can check app theme.
+  m_uiSettings.ColorValuesChanged([weakThis{Mso::WeakPtr(this)}](const auto &, const auto &) noexcept {
+    if (auto strongThis = weakThis.GetStrongPtr()) {
+      strongThis->InvokeInQueueStrong([strongThis]() noexcept { strongThis->OnColorValuesChanged(); });
+    }
+  });
+
+  m_inited.Set();
+}
+
+const char *AppearanceChangeListener::GetColorScheme() noexcept {
+  // Wait until we have an initial valid value from the UI thread before returning anything
+  m_inited.Wait();
+  return ToString(m_currentTheme);
+}
+
+const char *AppearanceChangeListener::ToString(ApplicationTheme theme) noexcept {
```

Comment:
>const char * [](start = 0, length = 12)

PCSTR since these are null terminated
(here and above)

Reply:
I'd rather use standard recognizable types than Win32 ones. I think people would assume this is null terminated if they saw the API.

Reply:
I found that I use the std::string_view in more and more cases like this. It provides string semantic, constexpr friendly, has bounds, and almost no overhead.

---
In reply to: [409752290](https://github.com/microsoft/react-native-windows/pull/4622#discussion_r409752290) [](ancestors = 409752290)

Reply:
The issue I've had with string_view is interop with c string APIs, since it doesn't guarantee a null terminator. A lot of libraries (I think even folly::dynamic) don't support string view currently, meaning we'd need to do extra allocations.

If I know I'm only going to use it for APIs I control I'd definitely go for it.